src/main/java/com/example/eventorganizer/user/UserRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.example.eventorganizer.user;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Repository
+@Transactional(readOnly = true)
+public interface UserRepository extends JpaRepository<User,Long> {
```

Comment:
There is no point in putting this annotation on repository. You need this annotation for service methods that use reporitory, in your case, on method loadUserByUsername or whole class.
https://www.baeldung.com/transaction-configuration-with-jpa-and-spring